### PR TITLE
[Catalog] make all example titles fit iPhone SE screen without clippi…

### LINKED ISSUE #3835: [Catalog] Make all example titles fit an iPhone SE screen without clipping
Definition of done:

- There are no examples whose title clips on an iPhone SE in the example list view.

---
#### Internal data
- Associated internal bug: [b/117179028](http://b/117179028)

### PR DESCRIPTION
**Pull request by fluxminds**
…ng (#3835)

1. Fixed truncated titles in injected AppBars .
2. Fixed a truncated title in a non injected example manually (TabBarIndicatorTemplateExample). Since this was the only one truncated title in non injected instances, not sure if we should handle it at a higher level for now. We can probably do it after the refactoring of AppBar navigation controller is done.

closes #3835

### CODE REVIEW

**Comment by jverkoey** [comment]
Ah! We've had a discussion about exposing this API on MDCNavigationBar in the past and the design team was hesitant to support it. Sending you a link off-line, but this particular change may require a quick chat with the design owner of Top App Bar.

File: catalog/MDCCatalog/MDCNodeListViewController.swift
Comment (andrewoverton) [inline catalog/MDCCatalog/MDCNodeListViewController.swift:583]: The variable you’re creating in line 581 should be lowercase, and it shouldn’t have the same name as its type. You can also get rid of the `self`. Something like this would be better:
`let appBarNavigationController = navigationController as? MDCAppBarNavigationController`
And maybe this is a matter of taste/preference, but in general, I think it’s best to minimize optional chaining. I think that’s what it’s called… I’m talking about this:
`let optional3 = optional1?.optional2?.optional3`
I would just replace these three lines with:
```Swift
    guard let appBarNavigationController = navigationController as? MDCAppBarNavigationController,
      let appBar = appBarNavigationController.appBarViewController(for: vc)
      else { return /* this is a good place to handle errors etc */ }
    appBar.navigationBar.adjustsFontSizeToFitWidth = true
```

**Review by romoore — COMMENTED** [review]
Could you please post before/after screenshots?

**Comment by romoore** [comment]
@jverkoey Could you please post the outcome of the discussion when you have it? The [Material guidelines article for AppBars](https://material.io/design/components/app-bars-top.html#anatomy) clearly states that shrinking text to fit is a "Don't".

File: components/NavigationBar/src/MDCNavigationBar.h
Comment (romoore) [inline components/NavigationBar/src/MDCNavigationBar.h:232]: Should this property also apply to the leading/trailing buttons?  Maybe rename to `titleAdjustsFontSizeToFitWidth`?

**Review by romoore — CHANGES_REQUESTED** [review]
Unless we get design approval (and an updated Material guidelines article), we should not make this change.

**Comment by jverkoey** [comment]
We're chatting with design on Friday.

**Comment by ianegordon** [comment]
@jverkoey could you give an update on the status of the discussions with design?

**Comment by jverkoey** [comment]
The open action item is to connect with Android and Web on what their approaches are for this problem so that we can aim for consistency.

**Comment by fluxminds** [comment]
Following up the discussion, Android team created an AutoResizeTextView object for handling auto sizing issue. From their perspective, it makes sense to add this API to adjust the font size, or we can create our version of auto resizing text view.

**Comment by romoore** [comment]
I'm more concerned about the current guidance not permitting text shrinking. It's possible Android and Web have done things incorrectly. What was the outcome of the discussion with designers?

**Comment by jverkoey** [comment]
@romoore Can you point to the guidance that states that we disallow text shrinking? If you've already seen this it will be helpful to point out where it is so that we can target changing it.

**Comment by romoore** [comment]
I mentioned it [in my original comment when I raised concerns about shrinking the text](https://github.com/material-components/material-components-ios/pull/4793#issuecomment-413204499). Here's a screenshot:

screen shot 2018-08-31 at 10 06 07 am

**Comment by jverkoey** [comment]
Awesome thank you!

**Comment by romoore** [comment]
This guidance aligns with the previous guidance on headers for full-screen dialogs (similar to how AppBars look/function, though different with respect to navigation):

https://material.io/archive/guidelines/components/dialogs.html#dialogs-full-screen-dialogs

> Full-screen dialog titles don’t use dynamic type.

I believe the general theme here is that view header titles and navigation element titles should not shrink or resize to fit. The design should account for long text by either truncating (where allowed) or placing the title elsewhere in the view.

**Comment by jverkoey** [comment]
@wenyuzhang666 this PR looks a bit stuck due to the need to get the spec updated. Do you mind if we close it for now and revisit once the spec work has landed?

**Comment by fluxminds** [comment]
@jverkoey I agree. I think the last thing I need to merge this PR is waiting for Javier's input on the design decision and update the spec (honestly I really hope I can directly link (@) designer on Github so they are included in the workflow), and then create a set of bugs for each platform. Should I close this one for now and wait for designer's input?

**Comment by jverkoey** [comment]
I think it's best to close this for now until we resolve the upstream design changes.

**Comment by fluxminds** [comment]
Closing this PR for now. Tracking note made in the original issue.

**Pull request closed without merging**
